### PR TITLE
fix(frontend): persist active room and rooms to LocalStorage

### LINKED ISSUE #1313: 🐛 [Bug] After reloading room page, there is no room
## 🐛 Bug
As a user, I might reload the room page for whatever reason and I still would like to see the same room. This is not the case - the page is just empty.

As a solution, I think we need to persist the activeRoomStore, probably using https://github.com/prazdevs/pinia-plugin-persistedstate.

(We might want to persist other stores as well.)

### PR DESCRIPTION
**Pull request by Bettelstab**
## 🍰 Pullrequest
Adds https://github.com/prazdevs/pinia-plugin-persistedstate and uses it to persist the stores for active room and rooms to LocalStorage.

### Issues
- fixes #1313

### CODE REVIEW

File: frontend/package-lock.json [hunk 1]
```diff
@@ -18735,6 +18735,7 @@
       "version": "3.2.1",
       "resolved": "https://registry.npmjs.org/pinia-plugin-persistedstate/-/pinia-plugin-persistedstate-3.2.1.tgz",
       "integrity": "sha512-MK++8LRUsGF7r45PjBFES82ISnPzyO6IZx3CH5vyPseFLZCk1g2kgx6l/nW8pEBKxxd4do0P6bJw+mUSZIEZUQ==",
+      "license": "MIT",
```
Comment (roschaefer) [inline frontend/package-lock.json:18738]: Unrelated?

File: frontend/src/stores/activeRoomStore.ts [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
-})
+    return {
+      activeRoom,
+      setActiveRoom,
+      getActiveRoom,
+    }
+  },
+  {
+    persist: true,
+  },
```
Comment (roschaefer) [inline frontend/src/stores/activeRoomStore.ts:24]: That's the only relevant change here.

Could we use a linter that enforces a certain indentaion?

There are two problems here:
1. It takes time and scrutiny to see the actual differences here, making code reviews harder.
2. @Bettelstab now owns these lines of code. This will make `git blame` much more difficult.

File: frontend/src/stores/roomsStore.ts [hunk 3]
```diff
@@ -18,55 +18,61 @@ export type Room = {
...
-})
+    return {
+      rooms,
+      setRooms,
+      getRooms,
+      isLoading,
+    }
+  },
+  {
+    persist: true,
```
Comment (roschaefer) [inline frontend/src/stores/roomsStore.ts:73]: Same here right? Some much changed code for such a small difference.

**Comment by Elweyn** [comment]
@roschaefer on stage1 it comes from two things:
- The refetch mechanismen gives us strange answers and can't update the store...
- The backend pulls the open rooms from BBB every 30 or 60 seconds so even after a refresh it can be that the backend database thinks that your room is still open with a person.

**Comment by Bettelstab** [comment]
@roschaefer 
You can test here:
 - https://stage1.dreammall.earth/ 
- Enter a table. E.g. click the button in the center and create table. You are redirected to /room
- Reload the page

So... as the unrelated changes are because of the linter, besides of the one line license thing, would you give me an approve? :)

**Review by roschaefer — APPROVED** [review]
> @roschaefer You can test here:
> 
>     * https://stage1.dreammall.earth/
> 
>     * Enter a table. E.g. click the button in the center and create table. You are redirected to /room
> 
>     * Reload the page
> 
> 
> So... as the unrelated changes are because of the linter, besides of the one line license thing, would you give me an approve? :)

It's exactly what I did. I will upload another screenshot:

image


I will click on "Approve" now because I see some strange behaviour. Feel free to merge if you think I got it now.

**Comment by Bettelstab** [comment]
@roschaefer It's not about the room list. It's about the actual room in the iframe.
